shared/actions/chat2/index.js

```diff
@@ -286,23 +286,50 @@ const onIncomingMessage = (incoming: RPCChatTypes.IncomingMessage, state: TypedS
         }
       }
     } else if (cMsg.state === RPCChatTypes.chatUiMessageUnboxedState.valid && cMsg.valid) {
-      const body = cMsg.valid.messageBody
+      const valid = cMsg.valid
+      const body = valid.messageBody
+      logger.info(`Got chat incoming message of messageType: ${body.messageType}`)
       // Types that are mutations
       switch (body.messageType) {
         case RPCChatTypes.commonMessageType.edit:
           if (body.edit) {
             actions.push(
               Chat2Gen.createMessageWasEdited({
                 conversationIDKey,
-                ...Constants.uiMessageEditToMessage(body.edit, cMsg.valid),
+                ...Constants.uiMessageEditToMessage(body.edit, valid),
               })
             )
           }
           break
         case RPCChatTypes.commonMessageType.delete:
           if (body.delete && body.delete.messageIDs) {
+            // check if the delete is acting on an exploding message
+            const messageIDs = body.delete.messageIDs
+            const messages = state.chat2.messageMap.get(conversationIDKey)
+            let isExplodeNow = false
+            if (messages) {
+              for (let i = 0; i < messageIDs.length; i++) {
+                const id = messageIDs[i]
+                const message = messages.find(msg => msg.id === id)
```

Comment:
you might want to actually use the map and see if the id is a key since i assume most of the time the id and ordinal will be the same if you didn't send it in this session and it'll be way quicker. if we have to do these lookups it might be faster to just keep a map. i worry about having to linearly go through

Reply:
Ok, I changed it to check the map first. In my testing `messages` usually has less than around 200 entries. Maybe it's a little premature to add an index, we could try this out and see if we notice hanging before a message transitions to `exploded by...`

Reply:
yeah but if you scroll up then it starts to grow and grow